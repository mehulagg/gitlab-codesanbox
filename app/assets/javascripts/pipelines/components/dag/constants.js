/* Error constants */
export const PARSE_FAILURE = 'parse_failure';
export const LOAD_FAILURE = 'load_failure';
export const UNSUPPORTED_DATA = 'unsupported_data';
export const DEFAULT = 'default';

/* Interaction handles */
export const IS_HIGHLIGHTED = 'dag-highlighted';
export const LINK_SELECTOR = 'dag-link';
export const NODE_SELECTOR = 'dag-node';

/* Annotation types */
export const ADD_NOTE = 'add';
export const REMOVE_NOTE = 'remove';
export const REPLACE_NOTES = 'replace';
