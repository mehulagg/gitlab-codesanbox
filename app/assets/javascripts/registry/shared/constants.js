import { s__, __ } from '~/locale';

export const FETCH_SETTINGS_ERROR_MESSAGE = s__(
  'ContainerRegistry|Something went wrong while fetching the cleanup policy.',
);

export const UPDATE_SETTINGS_ERROR_MESSAGE = s__(
  'ContainerRegistry|Something went wrong while updating the cleanup policy.',
);

export const UPDATE_SETTINGS_SUCCESS_MESSAGE = s__(
  'ContainerRegistry|Cleanup policy successfully saved.',
);

export const NAME_REGEX_LENGTH = 255;

export const ENABLED_TEXT = __('Enabled');
export const DISABLED_TEXT = __('Disabled');

export const ENABLE_TOGGLE_LABEL = s__('ContainerRegistry|Cleanup policy:');
export const ENABLE_TOGGLE_DESCRIPTION = s__(
  'ContainerRegistry|%{toggleStatus} - Tags matching the patterns defined below will be scheduled for deletion',
);

export const TEXT_AREA_INVALID_FEEDBACK = s__(
  'ContainerRegistry|The value of this input should be less than 256 characters',
);

export const EXPIRATION_INTERVAL_LABEL = s__('ContainerRegistry|Expiration interval:');
export const EXPIRATION_SCHEDULE_LABEL = s__('ContainerRegistry|Expiration schedule:');
export const KEEP_N_LABEL = s__('ContainerRegistry|Number of tags to retain:');
export const NAME_REGEX_LABEL = s__(
  'ContainerRegistry|Tags with names matching this regex pattern will %{italicStart}expire:%{italicEnd}',
);
export const NAME_REGEX_PLACEHOLDER = '.*';
export const NAME_REGEX_DESCRIPTION = s__(
  'ContainerRegistry|Wildcards such as %{codeStart}.*-test%{codeEnd} or %{codeStart}dev-.*%{codeEnd} are supported. To select all tags, use %{codeStart}.*%{codeEnd}',
);
export const NAME_REGEX_KEEP_LABEL = s__(
  'ContainerRegistry|Tags with names matching this regex pattern will %{italicStart}be preserved:%{italicEnd}',
);
export const NAME_REGEX_KEEP_PLACEHOLDER = '';
export const NAME_REGEX_KEEP_DESCRIPTION = s__(
  'ContainerRegistry|Wildcards such as %{codeStart}.*-master%{codeEnd} or %{codeStart}release-.*%{codeEnd} are supported',
);
