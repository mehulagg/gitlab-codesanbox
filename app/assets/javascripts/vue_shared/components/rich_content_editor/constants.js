import { __ } from '~/locale';

export const CUSTOM_EVENTS = {
  openAddImageModal: 'gl_openAddImageModal',
};

/* eslint-disable @gitlab/require-i18n-strings */
export const TOOLBAR_ITEM_CONFIGS = [
  { icon: 'heading', event: 'openHeadingSelect', classes: 'tui-heading', tooltip: __('Headings') },
  { icon: 'bold', command: 'Bold', tooltip: __('Add bold text') },
  { icon: 'italic', command: 'Italic', tooltip: __('Add italic text') },
  { icon: 'strikethrough', command: 'Strike', tooltip: __('Add strikethrough text') },
  { isDivider: true },
  { icon: 'quote', command: 'Blockquote', tooltip: __('Insert a quote') },
  { icon: 'link', event: 'openPopupAddLink', tooltip: __('Add a link') },
  { isDivider: true },
  { icon: 'list-bulleted', command: 'UL', tooltip: __('Add a bullet list') },
  { icon: 'list-numbered', command: 'OL', tooltip: __('Add a numbered list') },
  { icon: 'list-task', command: 'Task', tooltip: __('Add a task list') },
  { icon: 'list-indent', command: 'Indent', tooltip: __('Indent') },
  { icon: 'list-outdent', command: 'Outdent', tooltip: __('Outdent') },
  { isDivider: true },
  { icon: 'dash', command: 'HR', tooltip: __('Add a line') },
  { icon: 'table', event: 'openPopupAddTable', classes: 'tui-table', tooltip: __('Add a table') },
  { icon: 'doc-image', event: CUSTOM_EVENTS.openAddImageModal, tooltip: __('Insert an image') },
  { isDivider: true },
  { icon: 'code', command: 'Code', tooltip: __('Insert inline code') },
  { icon: 'doc-code', command: 'CodeBlock', tooltip: __('Insert a code block') },
];

export const EDITOR_TYPES = {
  markdown: 'markdown',
  wysiwyg: 'wysiwyg',
};

export const EDITOR_HEIGHT = '100%';

export const EDITOR_PREVIEW_STYLE = 'horizontal';

export const IMAGE_TABS = { UPLOAD_TAB: 0, URL_TAB: 1 };

export const MAX_FILE_SIZE = 2097152; // 2Mb
