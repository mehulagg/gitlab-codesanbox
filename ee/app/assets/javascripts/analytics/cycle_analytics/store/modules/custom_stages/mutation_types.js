export const SET_LOADING = 'SET_LOADING';
export const SET_STAGE_EVENTS = 'SET_STAGE_EVENTS';
export const SET_STAGE_FORM_ERRORS = 'SET_STAGE_FORM_ERRORS';
export const SET_FORM_INITIAL_DATA = 'SET_FORM_INITIAL_DATA';
export const SET_SAVING_CUSTOM_STAGE = 'SET_SAVING_CUSTOM_STAGE';
export const CLEAR_SAVING_CUSTOM_STAGE = 'CLEAR_SAVING_CUSTOM_STAGE';

export const HIDE_FORM = 'HIDE_FORM';
export const SHOW_CREATE_FORM = 'SHOW_CREATE_FORM';
export const SHOW_EDIT_FORM = 'SHOW_EDIT_FORM';
export const CLEAR_FORM_ERRORS = 'CLEAR_FORM_ERRORS';

export const RECEIVE_CREATE_STAGE_SUCCESS = 'RECEIVE_CREATE_STAGE_SUCCESS';
export const RECEIVE_CREATE_STAGE_ERROR = 'RECEIVE_CREATE_STAGE_ERROR';
