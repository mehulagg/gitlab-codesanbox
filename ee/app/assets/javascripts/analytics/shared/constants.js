export const dateFormats = {
  isoDate: 'yyyy-mm-dd',
  defaultDate: 'mmm d, yyyy',
  defaultDateTime: 'mmm d, yyyy h:MMtt',
};

export const scatterChartLineProps = {
  default: {
    type: 'line',
    showSymbol: false,
    // By default zlevel is 2 for all series types.
    // By increasing the zlevel to 3 we make sure that the trendline gets drawn in front of the dots in the chart.
    zlevel: 3,
  },
};

export const LAST_ACTIVITY_AT = 'last_activity_at';

export const SIMILARITY_ORDER = 'similarity';

export const DATE_RANGE_LIMIT = 180;

export const OFFSET_DATE_BY_ONE = 1;

export const NO_DRAG_CLASS = 'no-drag';

export const DATA_REFETCH_DELAY = 250;
