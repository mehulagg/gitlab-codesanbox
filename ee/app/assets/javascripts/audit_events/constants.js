import { __, s__ } from '~/locale';

import UserToken from './components/tokens/user_token.vue';
import MemberToken from './components/tokens/member_token.vue';
import ProjectToken from './components/tokens/project_token.vue';
import GroupToken from './components/tokens/group_token.vue';

const DEFAULT_TOKEN_OPTIONS = {
  operators: [{ value: '=', description: __('is'), default: 'true' }],
  unique: true,
};

// Due to the i18n eslint rule we can't have a capitalized string even if it is a case-aware URL param
/* eslint-disable @gitlab/require-i18n-strings */
const ENTITY_TYPES = {
  USER: 'User',
  AUTHOR: 'Author',
  GROUP: 'Group',
  PROJECT: 'Project',
};
/* eslint-enable @gitlab/require-i18n-strings */

export const AUDIT_FILTER_CONFIGS = [
  {
    ...DEFAULT_TOKEN_OPTIONS,
    icon: 'user',
    title: s__('AuditLogs|User Events'),
    type: 'user',
    entityType: ENTITY_TYPES.USER,
    token: UserToken,
  },
  {
    ...DEFAULT_TOKEN_OPTIONS,
    icon: 'user',
    title: s__('AuditLogs|Member Events'),
    type: 'member',
    entityType: ENTITY_TYPES.AUTHOR,
    token: MemberToken,
  },
  {
    ...DEFAULT_TOKEN_OPTIONS,
    icon: 'bookmark',
    title: s__('AuditLogs|Project Events'),
    type: 'project',
    entityType: ENTITY_TYPES.PROJECT,
    token: ProjectToken,
  },
  {
    ...DEFAULT_TOKEN_OPTIONS,
    icon: 'group',
    title: s__('AuditLogs|Group Events'),
    type: 'group',
    entityType: ENTITY_TYPES.GROUP,
    token: GroupToken,
  },
];

export const AVAILABLE_TOKEN_TYPES = AUDIT_FILTER_CONFIGS.map(token => token.type);
