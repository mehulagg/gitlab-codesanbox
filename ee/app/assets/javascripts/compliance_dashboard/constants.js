export const PRESENTABLE_APPROVERS_LIMIT = 2;

export const COMPLIANCE_TAB_COOKIE_KEY = 'compliance_dashboard_tabs';
