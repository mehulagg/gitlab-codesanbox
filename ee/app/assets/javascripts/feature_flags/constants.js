import { property } from 'lodash';
import { s__ } from '~/locale';

export const ROLLOUT_STRATEGY_ALL_USERS = 'default';
export const ROLLOUT_STRATEGY_PERCENT_ROLLOUT = 'gradualRolloutUserId';
export const ROLLOUT_STRATEGY_USER_ID = 'userWithId';
export const ROLLOUT_STRATEGY_GITLAB_USER_LIST = 'gitlabUserList';

export const PERCENT_ROLLOUT_GROUP_ID = 'default';

export const DEFAULT_PERCENT_ROLLOUT = '100';

export const ALL_ENVIRONMENTS_NAME = '*';

export const INTERNAL_ID_PREFIX = 'internal_';

export const fetchPercentageParams = property(['parameters', 'percentage']);
export const fetchUserIdParams = property(['parameters', 'userIds']);

export const NEW_VERSION_FLAG = 'new_version_flag';
export const LEGACY_FLAG = 'legacy_flag';

export const NEW_FLAG_ALERT = s__(
  'FeatureFlags|Feature Flags will look different in the next milestone. No action is needed, but you may notice the functionality was changed to improve the workflow.',
);

export const FEATURE_FLAG_SCOPE = 'featureFlags';
export const USER_LIST_SCOPE = 'userLists';
