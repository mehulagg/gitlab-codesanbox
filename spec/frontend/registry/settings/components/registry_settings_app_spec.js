import { shallowMount } from '@vue/test-utils';
import { GlAlert, GlSprintf, GlLink } from '@gitlab/ui';
import component from '~/registry/settings/components/registry_settings_app.vue';
import SettingsForm from '~/registry/settings/components/settings_form.vue';
import { createStore } from '~/registry/settings/store/';
import { SET_SETTINGS, SET_INITIAL_STATE } from '~/registry/settings/store/mutation_types';
import { FETCH_SETTINGS_ERROR_MESSAGE } from '~/registry/shared/constants';
import {
  UNAVAILABLE_FEATURE_INTRO_TEXT,
  UNAVAILABLE_USER_FEATURE_TEXT,
} from '~/registry/settings/constants';

import { stringifiedFormOptions } from '../../shared/mock_data';

describe('Registry Settings App', () => {
  let wrapper;
  let store;

  const findSettingsComponent = () => wrapper.find(SettingsForm);
  const findAlert = () => wrapper.find(GlAlert);

  const mountComponent = ({ dispatchMock = 'mockResolvedValue' } = {}) => {
    const dispatchSpy = jest.spyOn(store, 'dispatch');
    dispatchSpy[dispatchMock]();

    wrapper = shallowMount(component, {
      stubs: {
        GlSprintf,
      },
      mocks: {
        $toast: {
          show: jest.fn(),
        },
      },
      store,
    });
  };

  beforeEach(() => {
    store = createStore();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders', () => {
    store.commit(SET_SETTINGS, { foo: 'bar' });
    mountComponent();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('call the store function to load the data on mount', () => {
    mountComponent();
    expect(store.dispatch).toHaveBeenCalledWith('fetchSettings');
  });

  it('renders the setting form', () => {
    store.commit(SET_SETTINGS, { foo: 'bar' });
    mountComponent();
    expect(findSettingsComponent().exists()).toBe(true);
  });

  describe('the form is disabled', () => {
    beforeEach(() => {
      store.commit(SET_SETTINGS, undefined);
      mountComponent();
    });

    it('the form is hidden', () => {
      expect(findSettingsComponent().exists()).toBe(false);
    });

    it('shows an alert', () => {
      const text = findAlert().text();
      expect(text).toContain(UNAVAILABLE_FEATURE_INTRO_TEXT);
      expect(text).toContain(UNAVAILABLE_USER_FEATURE_TEXT);
    });

    describe('an admin is visiting the page', () => {
      beforeEach(() => {
        store.commit(SET_INITIAL_STATE, {
          ...stringifiedFormOptions,
          isAdmin: true,
          adminSettingsPath: 'foo',
        });
      });

      it('shows the admin part of the alert message', () => {
        const sprintf = findAlert().find(GlSprintf);
        expect(sprintf.text()).toBe('administration settings');
        expect(sprintf.find(GlLink).attributes('href')).toBe('foo');
      });
    });
  });

  describe('fetchSettingsError', () => {
    beforeEach(() => {
      mountComponent({ dispatchMock: 'mockRejectedValue' });
    });

    it('the form is hidden', () => {
      expect(findSettingsComponent().exists()).toBe(false);
    });

    it('shows an alert', () => {
      expect(findAlert().html()).toContain(FETCH_SETTINGS_ERROR_MESSAGE);
    });
  });
});
