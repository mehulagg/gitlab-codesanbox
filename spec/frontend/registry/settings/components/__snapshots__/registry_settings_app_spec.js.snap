// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registry Settings App renders 1`] = `
<div>
  <settings-form-stub />
</div>
`;
